src/prof.c

```diff
@@ -131,6 +132,28 @@ static int		prof_dump_fd;
 /* Do not dump any profiles until bootstrapping is complete. */
 static bool		prof_booted = false;
 
+
+typedef struct prof_prod_cons_data_s prof_prod_cons_data_t;
+typedef struct prof_prod_cons_key_s prof_prod_cons_key_t;
+
+struct prof_prod_cons_data_s {
+	uint64_t bytes;
+	uint64_t objs;
+};
+
+struct prof_prod_cons_key_s {
+	/* thread uids of the producer and consumer threads */
+	uint64_t prod;
+	uint64_t cons;
+};
+
+/*
+ * Maps producer-consumer pair (i.e. [prof_prod_cons_key_t]) to bytes and object
+ * counts (i.e. [prof_prod_cons_data_t]).
+ */
+static ckh_t		prod_cons;
+static malloc_mutex_t 	prod_cons_mtx;
```

Comment:
one small question I have about naming: is anything implied semantically by naming variables lock vs. mutex? It seems like both are used.